### PR TITLE
fix: add govet field alignment linting + changes to structs for better memory allocation

### PR DESCRIPTION
**Pull request by circa10a**
#### Proposed Changes ####

- Change linter to catch unoptimized structs that implement additional memory padding.
  - Implement fixes suggested by linter which is just rearranging the order of struct fields

#### Types of Changes ####

fix - better memory allocation when initializing structs

#### Verification ####

This is simply rearranging struct fields, so long as it compiles, changes are good.

#### Linked Issues ####

N/A

#### User-Facing Change ####

NONE

```release-note
Improve memory allocation when initializing structs
```

#### Further Comments ####

This isn't a huge change, but I thought it be useful to save on memory where possible in order to keep k3s as lightweight as can be.

### CODE REVIEW

File: .golangci.json [hunk 1]
```diff
@@ -35,4 +42,4 @@
 			}
 		]
 	}
-}
+}
```
Comment (briandowns) [inline .golangci.json:45]: Add new line back.

File: pkg/agent/netpol/network_policy_controller.go [hunk 2]
```diff
@@ -51,54 +51,42 @@ const (
...
 type networkPolicyInfo struct {
-	name        string
-	namespace   string
-	podSelector labels.Selector
-
-	// set of pods matching network policy spec podselector label selector
-	targetPods map[string]podInfo
-
-	// whitelist ingress rules from the network policy spec
+	podSelector  labels.Selector
```
Comment (briandowns) [inline pkg/agent/netpol/network_policy_controller.go:75]: Can you add back the comments that were removed?

**Comment by briandowns** [comment]
Did you perform a performance analysis before and after this optimization to gauge how much as actually gained?

**Comment by circa10a** [comment]
> Did you perform a performance analysis before and after this optimization to gauge how much as actually gained?

Not at runtime, but here are the before and after results of the byte savings report from the linter. By adding this to the linter going forward, it would ensure that new structs and modifications to existing stay optimal.

```
pkg/passwd/passwd.go:19:13: fieldalignment: struct with 16 pointer bytes could be 8 (govet)
type Passwd struct {
            ^
pkg/deploy/controller.go:65:14: fieldalignment: struct with 104 pointer bytes could be 88 (govet)
type watcher struct {
             ^
pkg/node/controller.go:32:14: fieldalignment: struct with 56 pointer bytes could be 48 (govet)
type handler struct {
             ^
pkg/servicelb/controller.go:84:14: fieldalignment: struct with 136 pointer bytes could be 128 (govet)
type handler struct {
             ^
pkg/agent/templates/templates.go:14:23: fieldalignment: struct with 32 pointer bytes could be 24 (govet)
type ContainerdConfig struct {
                      ^
pkg/agent/netpol/network_policy_controller.go:53:30: fieldalignment: struct with 272 pointer bytes could be 256 (govet)
type NetworkPolicyController struct {
                             ^
pkg/agent/netpol/network_policy_controller.go:78:24: fieldalignment: struct with 112 pointer bytes could be 104 (govet)
type networkPolicyInfo struct {
                       ^
pkg/agent/netpol/network_policy_controller.go:97:14: fieldalignment: struct with 56 pointer bytes could be 48 (govet)
type podInfo struct {
             ^
pkg/agent/netpol/network_policy_controller.go:105:18: fieldalignment: struct of size 112 could be 104 (govet)
type ingressRule struct {
                 ^
pkg/agent/netpol/network_policy_controller.go:115:17: fieldalignment: struct of size 112 could be 104 (govet)
type egressRule struct {
                ^
pkg/agent/netpol/network_policy_controller.go:130:16: fieldalignment: struct with 64 pointer bytes could be 56 (govet)
type endPoints struct {
               ^
pkg/agent/netpol/network_policy_controller_test.go:42:21: fieldalignment: struct with 24 pointer bytes could be 16 (govet)
type tNamespaceMeta struct {
                    ^
pkg/agent/netpol/network_policy_controller_test.go:213:22: fieldalignment: struct with 160 pointer bytes could be 152 (govet)
type tNetpolTestCase struct {
                     ^
pkg/agent/netpol/network_policy_controller_test.go:293:28: fieldalignment: struct with 40 pointer bytes could be 32 (govet)
type tNetPolConfigTestCase struct {
                           ^
pkg/authenticator/basicauth/basicauth_test.go:29:19: fieldalignment: struct of size 80 could be 72 (govet)
type testPassword struct {
                  ^
pkg/authenticator/basicauth/basicauth_test.go:47:26: fieldalignment: struct of size 160 could be 152 (govet)
        testCases := map[string]struct {
                                ^
pkg/authenticator/passwordfile/passwordfile_test.go:43:17: fieldalignment: struct with 40 pointer bytes could be 32 (govet)
        testCases := []struct {
                       ^
pkg/bootstrap/bootstrap_test.go:13:13: fieldalignment: struct with 40 pointer bytes could be 32 (govet)
        tests := []struct {
                   ^
pkg/clientaccess/token.go:43:11: fieldalignment: struct with 80 pointer bytes could be 72 (govet)
type Info struct {
          ^
pkg/daemons/config/types.go:28:11: fieldalignment: struct of size 992 could be 976 (govet)
type Node struct {
          ^
pkg/daemons/config/types.go:56:12: fieldalignment: struct of size 744 could be 736 (govet)
type Agent struct {
           ^
pkg/daemons/config/types.go:107:14: fieldalignment: struct of size 1024 could be 976 (govet)
type Control struct {
             ^
pkg/daemons/config/types.go:202:21: fieldalignment: struct with 888 pointer bytes could be 872 (govet)
type ControlRuntime struct {
```

**Pull request closed without merging**